disperser/cmd/apiserver/lib/config.go

```diff
@@ -92,16 +92,19 @@ func NewConfig(ctx *cli.Context) (Config, error) {
 		DisperserVersion: DisperserVersion(version),
 		AwsClientConfig:  aws.ReadClientConfig(ctx, flags.FlagPrefix),
 		ServerConfig: disperser.ServerConfig{
-			GrpcPort:                   ctx.GlobalString(flags.GrpcPortFlag.Name),
-			GrpcTimeout:                ctx.GlobalDuration(flags.GrpcTimeoutFlag.Name),
-			MaxConnectionAge:           ctx.GlobalDuration(flags.MaxConnectionAgeFlag.Name),
-			MaxConnectionAgeGrace:      ctx.GlobalDuration(flags.MaxConnectionAgeGraceFlag.Name),
-			MaxIdleConnectionAge:       ctx.GlobalDuration(flags.MaxIdleConnectionAgeFlag.Name),
-			PprofHttpPort:              ctx.GlobalString(flags.PprofHttpPort.Name),
-			EnablePprof:                ctx.GlobalBool(flags.EnablePprof.Name),
-			DisableGetBlobCommitment:   ctx.GlobalBool(flags.DisableGetBlobCommitment.Name),
-			SigningRateRetentionPeriod: ctx.GlobalDuration(flags.SigningRateRetentionPeriodFlag.Name),
-			SigningRatePollInterval:    ctx.GlobalDuration(flags.SigningRatePollIntervalFlag.Name),
+			GrpcPort:                       ctx.GlobalString(flags.GrpcPortFlag.Name),
+			GrpcTimeout:                    ctx.GlobalDuration(flags.GrpcTimeoutFlag.Name),
+			MaxConnectionAge:               ctx.GlobalDuration(flags.MaxConnectionAgeFlag.Name),
+			MaxConnectionAgeGrace:          ctx.GlobalDuration(flags.MaxConnectionAgeGraceFlag.Name),
+			MaxIdleConnectionAge:           ctx.GlobalDuration(flags.MaxIdleConnectionAgeFlag.Name),
+			PprofHttpPort:                  ctx.GlobalString(flags.PprofHttpPort.Name),
+			EnablePprof:                    ctx.GlobalBool(flags.EnablePprof.Name),
+			DisableGetBlobCommitment:       ctx.GlobalBool(flags.DisableGetBlobCommitment.Name),
+			SigningRateRetentionPeriod:     ctx.GlobalDuration(flags.SigningRateRetentionPeriodFlag.Name),
+			SigningRatePollInterval:        ctx.GlobalDuration(flags.SigningRatePollIntervalFlag.Name),
+			DisperserId:                    uint32(ctx.GlobalUint64(flags.DisperserIdFlag.Name)),
+			TolerateMissingAnchorSignature: ctx.GlobalBool(flags.TolerateMissingAnchorSignatureFlag.Name),
+			ChainId:                        nil, // Will be populated after eth client is created
```

Comment:
IMO, it's a code smell to have config that isn't populated when the config is loaded. Not necessarily a blocker if there is no other clean way to do it.